### PR TITLE
New Breadcrumbs component

### LINKED ISSUE #1488: `margin` values not following the guidelines
**Describe the bug**
Our current values for both `padding` and `margin` props are not following the actual guidelines. We must stick to our principles so we need to update them.

**Screenshots**
Implementation

image

[Spacing principles](https://developer.dxc.com/halstack/next/principles/spacing/#component-spacing-tokens)

image

### PR DESCRIPTION
**Pull request by GomezIvann**
**Checklist**

- [x] The build process is done without errors. All tests pass in the `/lib` directory.
- [x] Self-reviewed the code before submitting.
- [x] Meets accessibility standards.
- [x] Added/updated documentation to `/website` as needed.
- [x] Added/updated tests as needed.

**Description**
The breadcrumbs component represents a list of links showing the location of the current page in the navigational hierarchy.

**Requirements**
To be able to implement this component, some extra requirements were needed:
- The `@types/jest` library was added so we can start to name test files with the extension `.tsx`, which is more correct.
- Updated spacing values in the design system to match the [current specifications](https://developer.dxc.com/halstack/next/principles/spacing/#component-spacing-tokens).

**Design**
[UXPin Preview](https://preview.uxpin.com/1b969481ff43ae3bcd40cdabe172b8d1784638a4#/pages/165667295/specification/no-panels)

**Screenshot**

image

image

**Closes #1488**

### CODE REVIEW

**Comment by Mil4n0r** [comment]
Currently, accessibility tests are not passing due to the same reason they weren't initally passing for `Header` and `Footer`.

We should create a `disabledRules.js` file inside lib/test/accessibility/specific/breadcrumbs similar to the one existing in the header but with only the `"landmark-unique"` rule as exception.

It should be added to both the Story file and the Breadcrumbs.accessibility.test that should be created before merging this PR.

File: lib/setupJestAxe.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
-import { toHaveNoViolations } from "jest-axe";
+import { toHaveNoViolations } from 'jest-axe';
```
Comment (Jialecl) [inline lib/setupJestAxe.js:1]: We don't use `' '` in imports.

File: lib/src/breadcrumbs/Breadcrumbs.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import DxcFlex from "../flex/Flex";
+
+const DxcBreadcrumbs = ({
+  ariaLabel = "Breadcrumbs",
+  items,
+  itemsBeforeCollapse = 4,
+  onItemClick,
+  showRoot = true,
+}: BreadcrumbsProps) => {
+  const colorsTheme = useTheme();
```
Comment (Jialecl) [inline lib/src/breadcrumbs/Breadcrumbs.tsx:20]: colorsTheme is no longer in use.

File: lib/src/breadcrumbs/Breadcrumbs.accessibility.test.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+describe("Breadcrumbs component accessibility tests", () => {
+  it("Should not have basic accessibility issues", async () => {
+    const { container } = render(<DxcBreadcrumbs items={items} itemsBeforeCollapse={3} ariaLabel="example" />);
+    const results = await axe(container, disabledRules);
+    expect(results).toHaveNoViolations();
+  });
+
+  it("Should not have basic accessibility issues without root", async () => {
+    const { container } = render(
+      <DxcBreadcrumbs items={items} itemsBeforeCollapse={3} ariaLabel="example" showRoot={false} />
```
Comment (Jialecl) [inline lib/src/breadcrumbs/Breadcrumbs.accessibility.test.tsx:42]: Should we consider accessibility tests for breadcrumbs without collapse?

File: lib/setupJestAxe.js (hunk 1 above)
Comment (GomezIvann) [reply lib/setupJestAxe.js]: Updated!

File: lib/src/breadcrumbs/Breadcrumbs.accessibility.test.tsx (hunk 3 above)
Comment (GomezIvann) [reply lib/src/breadcrumbs/Breadcrumbs.accessibility.test.tsx:46]: Done!

**Review by Jialecl — APPROVED** [review]

**Review by Jialecl — DISMISSED** [review]

**Review by Jialecl — APPROVED** [review]

**Pull request merged**